pyproject.toml

```
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=45", "wheel", "setuptools_scm>=6.2"]
+requires = ["setuptools>=45", "wheel", "setuptools_scm>=6.2,<8.2"]
```

I'd prefer to upversion this like I did on april_vision last year.
```suggestion
requires = ["setuptools>=60", "wheel", "setuptools_scm>=8,<9"]
```

---

Unfortunately that change wouldn't be compatible with the current deployment mechanism, which is coupled to `setuptools` from the distro and which is version `59.6.0`. The better fix is to put the pystream in its own virtualenv (https://github.com/PeterJCLaw/srcomp-puppet/issues/31), however this is intended as a temporary workaround.

---

Then we need to restrict this to `setuptools-scm<8`.
```suggestion
requires = ["setuptools>=45", "wheel", "setuptools_scm>=6.2,<8"]
```

---

We can do that, though it doesn't seem necessary -- I have tested the change as it stands and it seems to work.

The specific change which triggers the issue we're working around here is the change to the _runtime_ dependency (from `setuptools-scm`) on `setuptools`, which only appeared in 8.2 (or at least is only declared in 8.2), rather than the build time dependency (which does indeed change in 8.0).

Is there something I'm missing?